### DISCUSSION TITLE
Use GOOGLE_APPLICATION_CREDENTIALS path vs. json for PubSub

### DISCUSSION THREAD

**Discussion by delormej** [kedacore/keda#1188] [Ideas]
As described in the getting started guide for the Google SDK (https://cloud.google.com/docs/authentication/getting-started), typically applications set an env variable that points to the json *file* as opposed to actually storing the json string in the env variable.  It seems the scaler for Google PubSub is expected the json string rather than the file location.  It would be great if it were consistent with the SDK usage to avoid duplicate work.  Maybe I'm wrong in thinking this, so before I start working on a PR here, just wondering if anyone has opposing thoughts on this?

**Comment by turbaszek** [comment]
+1 for pointing to json service key file instead of keeping the content in env variable, that's how it's usually done

**Comment by tomkerkhove** [comment]
Tracking in https://github.com/kedacore/keda/issues/1192

**Comment by zroubalik** [comment]
@delormej is this something you are willing to contribute?

**Reply by delormej** [reply]
Yes, happy to contribute this.
